app/src/main/java/com/babylon/wallet/android/presentation/transaction/TransactionReviewViewModel.kt

```diff
@@ -261,10 +268,17 @@ class TransactionReviewViewModel @Inject constructor(
     }
 
     fun dismissTerminalErrorDialog() {
-        (state.value.error?.error as? DappRequestException)?.let { exception ->
-            viewModelScope.launch { submit.onDismiss(exception) }
-        }
         _state.update { it.copy(error = null) }
+
+        viewModelScope.launch {
+            val error = state.value.error?.error
+            if (error is DappRequestException) {
+                submit.onDismiss(error)
+            } else {
+                incomingRequestRepository.requestHandled(args.interactionId)
+                sendEvent(Event.Dismiss)
```

Comment:
tbh i remember having some issues in the past with the order of those two but I can't remember what was exactly and if it was for transaction review screen or for the dapplogin screens.

is there any way we can test this?
Did you have the chance to test it?

Reply:
Yeap, I tested this, you can test it too by following these steps: disable 3rd party deposits for an account, go to another account settings with assets and choose to delete it -> when asked to move assets to another account choose the one with the 3rd party deposits disallowed and tap continue -> you're going to see the tx review screen with an error -> when tapping it it goes through this code which dismisses the screen and marks the request as handled.

Reply:
Although I agree that it would make more sense first to mark the request as handled and then sending the dismiss event.